databuilder/extractor/dashboard/mode_analytics/mode_dashboard_queries_extractor.py

```diff
@@ -0,0 +1,93 @@
+
+    """
+
+    def init(self, conf):
+        # type: (ConfigTree) -> None
+        self._conf = conf
+
+        restapi_query = self._build_restapi_query()
+        self._extractor = ModeDashboardUtils.create_mode_rest_api_extractor(
+            restapi_query=restapi_query,
+            conf=self._conf
+        )
+
+        # Constructing URL using several ID via TemplateVariableSubstitutionTransformer
+        transformers = []
+        variable_substitution_transformer = TemplateVariableSubstitutionTransformer()
+        variable_substitution_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf,
+                                        variable_substitution_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict({FIELD_NAME: 'url',
+                                         TEMPLATE: 'https://app.mode.com/{organization}'
+                                                   '/reports/{dashboard_id}/queries/{query_id}'})))
+
+        transformers.append(variable_substitution_transformer)
+
+        dict_to_model_transformer = DictToModel()
+        dict_to_model_transformer.init(
+            conf=Scoped.get_scoped_conf(self._conf, dict_to_model_transformer.get_scope()).with_fallback(
+                ConfigFactory.from_dict(
+                    {MODEL_CLASS: 'databuilder.models.dashboard.dashboard_query.DashboardQuery'})))
+        transformers.append(dict_to_model_transformer)
+
+        self._transformer = ChainedTransformer(transformers=transformers)
+
+    def extract(self):
+        # type: () -> Any
+
+        record = self._extractor.extract()
+        if not record:
+            return None
+
+        return self._transformer.transform(record=record)
+
+    def get_scope(self):
+        # type: () -> str
+        return 'extractor.mode_dashboard_query'
+
+    def _build_restapi_query(self):
+        """
+        Build REST API Query. To get Mode Dashboard last execution, it needs to call three APIs (spaces API, reports
+        API, and queries API) joining together.
+        :return: A RestApiQuery that provides Mode Dashboard execution (run)
+        """
+        # type: () -> RestApiQuery
+
+        spaces_query = ModeDashboardUtils.get_spaces_query_api(conf=self._conf)
+        params = ModeDashboardUtils.get_auth_params(conf=self._conf)
+
+        # Reports
+        # https://mode.com/developer/api-reference/analytics/reports/#listReportsInSpace
+        url = 'https://app.mode.com/api/{organization}/spaces/{dashboard_group_id}/reports'
+        json_path = '(_embedded.reports[*].token)'
+        field_names = ['dashboard_id']
+        reports_query = RestApiQuery(query_to_join=spaces_query, url=url, params=params,
+                                     json_path=json_path, field_names=field_names, skip_no_result=True)
+
+        queries_url_template = 'https://app.mode.com/api/{organization}/reports/{dashboard_id}/queries'
```

Comment:
and this one as well? how to cope with the different json response?

Reply:
Yes, this one provides different JSON response. I think I need to take a look at their documentation to figure out the next step. For now, token, and url is in the same place for the queries (as far as I've seen) and this should unblock us. WDYT?